flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/JobAutoScalerImpl.java

```diff
@@ -113,7 +115,13 @@ public boolean scale(FlinkResourceContext<? extends AbstractFlinkResource<?, ?>>
                     scalingExecutor.scaleResource(resource, autoScalerInfo, conf, evaluatedMetrics);
             autoScalerInfo.replaceInKubernetes(kubernetesClient);
             return specAdjusted;
-        } catch (Exception e) {
+        } catch (Throwable e) {
+            eventRecorder.triggerEvent(
+                    resource,
+                    EventRecorder.Type.Warning,
+                    EventRecorder.Reason.AutoscalerError,
+                    EventRecorder.Component.Operator,
+                    e.getMessage());
```

Comment:
if an NPE occurs, `e.gestMessage()` will be null. We need to make sure that `triggerEvent` and the downstream caller are not throwing NPE because of this

Reply:
Shouldn't be a problem and we do this in other places as well. I will add a simple test even if unrelated to the change.